### PR TITLE
[Release 1.5.0] Add new PowerShell Cmdlet for ResourceManager\MachineLearning

### PR DESCRIPTION
**Pull request by NonStatic2014**
Change Log:
- Azure Machine Learning 
  - [Preview] New cmdlets to manage Azure Machine Learning Web Services
    - New-AzureRmMlWebService
    - Get-AzureRmMlWebService
    - Remove-AzureRmMlWebService
    - Update-AzureRmMlWebService
    - Get-AzureRmMlWebServiceKeys
    - Import-AzureRmMlWebService
    - Export-AzureRmMlWebService

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@NonStatic2014__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (bohu). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by NonStatic2014** [comment]
Original pull request: https://github.com/Azure/azure-powershell/pull/2288

**Comment by NonStatic2014** [comment]
On-demand build passed: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/873/

File: src/ResourceManager/MachineLearning/AzureRM.MachineLearning.psd1 [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+DotNetFrameworkVersion = '4.0'  
+  
+# Minimum version of the common language runtime (CLR) required by this module  
+CLRVersion='4.0'  
+  
+# Processor architecture (None, X86, Amd64, IA64) required by this module  
+ProcessorArchitecture = 'None'  
+  
+# Modules that must be imported into the global environment prior to importing this module  
+RequiredModules = @( @{ ModuleName = 'AzureRM.Profile'; ModuleVersion = '1.0.8'})
```
Comment (hovsepm) [inline src/ResourceManager/MachineLearning/AzureRM.MachineLearning.psd1]: change this to 1.0.9. If this is the only change - no need to run another on-demand.

File: src/ResourceManager/MachineLearning/Commands.MachineLearning/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (hovsepm) [inline src/ResourceManager/MachineLearning/Commands.MachineLearning/Properties/AssemblyInfo.cs]: AssemblyVersion and AssemblyFileVersion needs to have the same value as AzureRM.MAchineLEarning.psd1 Module version is => 0.9.0

File: src/ResourceManager/MachineLearning/Commands.MachineLearning/Exception/ErrorResponseMessageException.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        /// <param name="errorMessage">The error response message.</param>
+        /// <param name="innerException">Optional. The inner exception.</param>
+        public ErrorResponseMessageException(HttpStatusCode httpStatus, ErrorResponseMessage errorResponseMessage, string errorMessage, Exception innerException = null)
+            : base(errorMessage, innerException)
+        {
+            if (((int)httpStatus >= 200 && (int)httpStatus <= 299) || (int)httpStatus == 304)
+            {
+                throw new ArgumentException(
+                    message: "The error response message exception should not be used for successful http response messages.",
+                    paramName: "httpStatus");
```
Comment (hovsepm) [inline src/ResourceManager/MachineLearning/Commands.MachineLearning/Exception/ErrorResponseMessageException.cs]: please put text string to Resources and use if from there.

**Comment by markcowl** [comment]
@NonStatic2014 This looks good, 3 required changes:

(1) Change the reference to AzureRM.Profile as specified by hovsepm
(2) Add you module to tools\AzureRM\AzureRM.psd1
(3) regenerate the installer wxi file using the instructions here: https://github.com/Azure/azure-powershell/wiki/Microsoft-Azure-PowerShell-Developer-Guide#updating-the-installer

**Comment by hovsepm** [comment]
@NonStatic2014 after addressing all the comments please run sing job and post the link to it here http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/

**Comment by NonStatic2014** [comment]
Kicked off the following build:
  - Sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/446/
  - On-demand build: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/875/

**Comment by NonStatic2014** [comment]
On-demand build passed.
Sign job failed with below error. Investigating....

13:37:58   D:\workspace\powershell-sign\build.proj(220,5): error : The assembly "D:\workspace\powershell-sign\src\Package\Release\ResourceManager\AzureResourceManager\AzureRM.MachineLearning\Microsoft.Azure.Commands.MachineLearning.dll" had the strong name token of "0745f5a326041567", but was expected to have the token "31bf3856ad364e35"

**Comment by NonStatic2014** [comment]
Sign job passed: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/447

**Pull request merged**